runtime-modules/forum/src/lib.rs

```diff
@@ -1400,12 +1400,12 @@ impl<T: Config> Module<T> {
             };
 
             <PostById<T>>::insert(thread_id, new_post_id, new_post);
-        }
-
-        let mut thread = <ThreadById<T>>::get(category_id, thread_id);
-        thread.number_of_posts = thread.number_of_posts.saturating_add(1);
 
-        <ThreadById<T>>::mutate(category_id, thread_id, |value| *value = thread);
+            <ThreadById<T>>::mutate(category_id, thread_id, |thread| {
+                // non editable post should leave the counter untouched
+                thread.number_of_posts = thread.number_of_posts.saturating_add(1);
```

Comment:
The filed should be renamed as suggested in https://github.com/Joystream/joystream/issues/4175#issuecomment-1215436306. Perhaps `number_of_mutable_posts`?

Reply:
renamed it to `number_of_editable_posts`